src/west/__init__.py

```diff
@@ -1,5 +0,0 @@
-# Copyright 2018 Open Source Foundries Limited.
```

Comment:
> At the moment, `west` is just a small helper application at least. Maybe some of the packages will get reused later.

Right, that's what we're planning for, so we need to keep west importable. Only the `commands` subpackage contains the commands. Do you remember discussing this in the kickoff meetings?

Reply:
I think you might be confusing the application with the packages it happens to be using. The application itself does not need to be importable.

Reply:
I'm not sure what distinction you are trying to make. Why should I not be able to import west?

Reply:
Hmm... think I might've been confused by the directory structure.

Not saying we have to change it, but I think this directory might illustrate how I imagined it (at the top level of the repository, for example):

```
west.py           ("main" executable. Uses the West library.)
bootstrap.py      (could reuse some parts of West as well)
west/__init__.py  (the West package)
west/...
```

If the application is bundled inside the library at the moment, then I get it.

Sorry if you've already told me, but what other stuff besides `west.py` and `bootstrap.py` would need to use the West library?

Reply:
Guess I'm confused about what we're making. Is it a library or an application? Will the library be used outside the application, so that e.g. some other application would be using the `runners/` classes?

Reply:
So from my point of view it should be both. An application that can be run directly from the command line, and also a library that can be imported and invoked directly by other scripts. So if `west` as a `sync` command, I need to be able both to type `west sync` from the command line but also to `import west` and then `west.sync()` from a script. That was my understanding at least.

Reply:
@carlescufi 
Yeah, that'd make sense, if we need it to be a library as well.

Not sure what the public interface of the library is at the moment, and how much of it is generally useful. If we want to make it a proper library, we should define what the interface is, and should probably follow some conventions like prefixing non-public stuff with `_` (prevents it from showing up in documentation, and affects e.g. `from west import *`).

Reply:
Will the library part ever be used by third-party applications not written within the Zephyr project? In that case, we need to at least do the following:

  1. Define the public API of the library (preferably prefixing internal stuff with `_`)

  2. Maintain API compatibility between versions

If the public API is "everything", then it's more of a grab bag of code than a library, and maintaining API compatibility will be extremely hard.

Going on personal experience, maintaining a library used by third parties is a ton more work than just having e.g. a shared codebase. If we don't need a library, we should definitely avoid it.